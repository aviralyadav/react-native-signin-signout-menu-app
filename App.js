/**
 *
 * @format
 * @flow
 */

import React, { Component } from "react";
import { createStackNavigator, createAppContainer, createSwitchNavigator } from "react-navigation";

import LoginScreen from './src/LoginScreen';
import DashboardScreen from "./src/DashboardScreen";
import NewFormScreen from "./src/NewFormScreen";
import SignUpScreen from "./src/SignUpScreen";
import ProfileScreen from "./src/ProfileScreen";
import AuthLoadingScreen from "./AuthLoadingScreen";

const AppStack = createStackNavigator({ 
  Dashboard: DashboardScreen,
  NewForm: NewFormScreen,
  Profile: ProfileScreen });
const AuthStack = createStackNavigator({ SignIn: LoginScreen });

export default createAppContainer(createSwitchNavigator(
  {
    AuthLoading: AuthLoadingScreen,
    App: AppStack,
    Auth: AuthStack,
  },
  {
    initialRouteName: 'AuthLoading',
  }
));

// const AppNavigator = createStackNavigator(
  //   {
  //   Home: LoginScreen,
  //   Dashboard: DashboardScreen,
  //   NewForm: NewFormScreen,
  //   SignUp: SignUpScreen,
  //   Profile: ProfileScreen
  // },
  // {
  //   initialRouteName: 'Home'
  // }
  // );
  

// export default createAppContainer(AppNavigator);

