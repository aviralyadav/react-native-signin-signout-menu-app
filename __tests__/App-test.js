/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { View, Text, TouchableHighlight } from 'react-native';

let props = {
  route: {
    uiOptions: {
      itemImage:
        "https://images.unsplash.com/photo-1533923156502-be31530547c4?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
    }
  },
  uiOptions: {
    privacyURL: "https://myaccount.google.com/privacycheckup",
    tosURL: "https://termsfeed.com/blog/examples-user-agreements/"
  },
  itemDetails: {
    itemName: "Football Game 1",
    internetName: "Stanford Cardinal Football",
    eventName: null,
    startDate: "2016-05-20T20:30:00Z",
    endDate: "2016-05-22T20:30:00Z",
    timeZone: "PST8PDT",
    image:
      "https://images.unsplash.com/photo-1533923156502-be31530547c4?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80",
    PrivacyPolicy: "https://myaccount.google.com/privacycheckup",
    TermsOfUse: "https://termsfeed.com/blog/examples-user-agreements/",
    details: {
      facility: {
        name: "Avalos Stadium",
        address1: "5171 California Avenue, Suite 200",
        address2: null,
        address3: null,
        address4: null,
        addressLocality: "Irvine",
        addressRegion: "CA",
        postalCode: "92617",
        addressCountry: "US",
        timeZone: "PST8PDT"
      }
    },
    displayTimes: {
      long: "2016-05-20T20:30:00Z"
    }
  }
};

it('renders correctly', () => {
  renderer.create(<App />);
});

// it('renders snapshot correctly', () => {
//   const tree = renderer.create(<App />).toJSON();
//   expect(tree).toMatchSnapshot();
// });

const shoppingList = [
  'diapers',
  'kleenex',
  'trash bags',
  'paper towels',
  'beer',
];

it('the shopping list has beer on it', () => {
  expect(shoppingList).toContain('beer');
});

it('renders the ActivityIndicator component', () => {
  const ActivityIndicator = require('ActivityIndicator');
  const tree = renderer
    .create(<ActivityIndicator animating={true} size="small" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('<App>', () => {
  it('updates number when clicked', () => {
    const inst = renderer.create(<App />);
    
    const button = inst.root.findByType(TouchableHighlight);
    const text = inst.root.findAllByType(Text);
    // default state
    // expect(text.props.children).toBe(2);
    // or
    expect(inst.root.instance.state.number).toBe(1);
    button.props.onPress();
    // state was updated by the button press
    // console.log(text.length(6));
    expect(text.props.children).toBe(2);
  });
});