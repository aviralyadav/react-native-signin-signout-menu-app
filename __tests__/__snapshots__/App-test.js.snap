// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#F5FCFF",
        "flex": 1,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": "red",
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      Welcome to React Native!
    </Text>
    <Text
      style={
        Object {
          "backgroundColor": "green",
          "color": "#333333",
          "textAlign": "center",
        }
      }
    >
      To get started, edit App.js
    </Text>
    <Text
      style={
        Object {
          "backgroundColor": "yellow",
          "color": "#333333",
          "marginBottom": 5,
          "textAlign": "center",
        }
      }
    >
      Press Cmd+R to reload,
Cmd+D or shake for dev menu
    </Text>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  >
    <View
      style={
        Object {
          "marginTop": 22,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "",
          }
        }
      >
        <Text>
          Hello World!
        </Text>
        <Text>
          Hello World! how are you , how is going all.
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <Text
            style={null}
          >
            Hide Modal
          </Text>
        </View>
      </View>
    </View>
  </Modal>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text
      style={null}
    >
      Show Modal
    </Text>
  </View>
</View>
`;

exports[`renders the ActivityIndicator component 1`] = `
<ActivityIndicator
  animating={true}
  color="#999999"
  hidesWhenStopped={true}
  size="small"
/>
`;
